plugins/docker/cmd.ml

```diff
@@ -13,7 +13,7 @@ let compose ~docker_context args =
   "", Array.of_list ("docker-compose" :: context_args docker_context @ args)
 
 let login ?config ~docker_context user =
-  docker ?config ~docker_context ["login"; "--password-stdin"; "--username"; user]
+  docker ?config ~docker_context (["login"; "--password-stdin"; "--username"] @ String.split_on_char '@' user)
```

Comment:
Why expose this as `user@host` rather than as a new optional paramter `?host`?

My question here is motivated by the principle that it is generally preferable to to work on fully parsed, structured data, rather than encoding compound data into strings.

`pull` and `push` could also be extended with a `?host` argument, in a change that should be none breaking for almost all use cases.

WDYT?